ethereum/gateway-eth-ts/src/utils/signer.ts

```diff
@@ -44,3 +45,113 @@ export const mnemonicSigner = function (
 
   return signer;
 };
+
+type Input = {
+  from: string;
+  to: string;
+  data: string;
+};
+
+const EIP712Domain = [
+  { name: "name", type: "string" },
+  { name: "version", type: "string" },
+  { name: "chainId", type: "uint256" },
+  { name: "verifyingContract", type: "address" },
+];
+
+const ForwardRequest = [
+  { name: "from", type: "address" },
+  { name: "to", type: "address" },
+  { name: "value", type: "uint256" },
+  { name: "gas", type: "uint256" },
+  { name: "nonce", type: "uint256" },
+  { name: "data", type: "bytes" },
+];
+
+export interface EIP712Message {
+  [key: string]: any;
+}
+
+interface SignedMetaTxRequest {
+  signature: string;
+  request: EIP712Message;
+}
+
+interface EIP712TypeProperty {
+  name: string;
+  type: string;
+}
+
+interface EIP712Types {
+  [name: string]: ReadonlyArray<EIP712TypeProperty>;
+}
+
+interface EIP712Domain {
+  name?: string | undefined;
+  version?: string | undefined;
+  chainId?: string | number | undefined;
+  verifyingContract?: string | undefined;
+  salt?: string | undefined;
+}
+
+export interface EIP712TypedData {
+  types: EIP712Types;
+  primaryType: string;
+  domain: EIP712Domain;
+  message: EIP712Message;
+}
+
+const getMetaTxTypeData = (
+  chainId: number,
+  verifyingContract: string
+): Omit<EIP712TypedData, "message"> => ({
+  types: {
+    EIP712Domain,
+    ForwardRequest,
+  },
+  domain: {
+    name: "MinimalForwarder",
+    version: "0.0.1",
+    chainId,
+    verifyingContract,
+  },
+  primaryType: "ForwardRequest",
+});
+
+async function signTypedData(signer: Wallet, data: EIP712TypedData) {
+  const types = { ForwardRequest } as Record<string, Array<TypedDataField>>;
+  return signer._signTypedData(data.domain, types, data.message);
+}
+
+const buildRequest = async (
+  forwarder: Contract,
```

Comment:
Can we not build an Interface for the forwarder Contract (as with the other contracts), to make the TS a little nicer?